Stablize fcos v1_6,  base v0_6, and bump all other exp to latest (#566)
coreos/butane: docs/upgrading-fcos.md

parent: Upgrading configs
nav_order: 1
---

> # Upgrading Fedora CoreOS configs
> 
> Occasionally, changes are made to Fedora CoreOS Butane configs (those that specify `variant: fcos`) that break backward compatibility. While this is not a concern for running machines, since Ignition only runs one time during first boot, it is a concern for those who maintain configuration files. This document serves to detail each of the breaking changes and tries to provide some reasoning for the change. This does not cover all of the changes to the spec - just those that need to be considered when migrating from one version to the next.
> 
> {: .no_toc }
> 
> 1. TOC
> {:toc}
> 

Reviewer: Not needed
```suggestion
```

## From Version 1.5.0 to Version 1.6.0

> There are no breaking changes between versions 1.4.0 and 1.5.0 of the `fcos` configuration specification. Any valid 1.4.0 configuration can be updated to a 1.5.0 configuration by changing the version string in the config.

Reviewer: ```suggestion
There are no breaking changes between versions 1.5.0 and 1.6.0 of the `fcos` configuration specification. Any valid 1.5.0 configuration can be updated to a 1.65.0 configuration by changing the version string in the config.
```

The following is a list of notable new features.

### Luks CEX support

> The `luks` sections in `storage` and `boot_device` gained a `cex` field. If enabled, this will configure an encrypted root filesystem on a s390x using IBM Crypto Express (CEX) card.

Reviewer: ```suggestion
The `luks` sections in `storage` and `boot_device` gained a `cex` field. If enabled, this will configure an encrypted root filesystem on a s390x system using IBM Crypto Express (CEX) card.
```

<!-- butane-config -->
```yaml
variant: fcos
version: 1.6.0
boot_device:
  layout: s390x-eckd
  luks:
    device: /dev/dasda
    cex:
      enabled: true

> ```

Reviewer: This should go below the following lines:
```
1. TOC
{:toc}
```

1. TOC
{:toc}

## From Version 1.5.0 to Version 1.6.0

There are no breaking changes between versions 1.5.0 and 1.6.0 of the `fcos` configuration specification. Any valid 1.5.0 configuration can be updated to a 1.6.0 configuration by changing the version string in the config.

The following is a list of notable new features.

> 

Author: Shoot, ok thank you

> 

Author: added docs for the layouts in the update notes

> ### Luks CEX support

Reviewer: ```suggestion
### LUKS CEX support
```
